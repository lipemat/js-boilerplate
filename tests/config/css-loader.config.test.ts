const {getLocalIdent} = require( '../../helpers/css-classnames' );

// Change this variable during tests.
let mockShortCssEnabled = false;

// Change the result of the getPackageConfig function so we can change shortCssClasses.
jest.mock( '../../helpers/package-config.ts', () => ( {
	...jest.requireActual( '../../helpers/package-config.ts' ),
	getPackageConfig: () => ( {
		...jest.requireActual( '../../helpers/package-config.ts' ),
		// Change this variable during the test.
		shortCssClasses: mockShortCssEnabled,
	} ),
} ) );
// Change the result of the getLocalIdent function to something we can verify.
jest.mock( '../../helpers/css-classnames.js', () => ( {
	...jest.requireActual( '../../helpers/css-classnames.js' ),
	getLocalIdent: jest.fn().mockReturnValue( '__TEST_CSS__' ),
} ) );


beforeEach( () => {
	jest.resetModules();
} );

afterEach( () => {
	process.env.NODE_ENV = 'test';
	mockShortCssEnabled = false;
} );

describe( 'css-loader.config.test.ts', () => {
	test( 'Develop config', () => {
		const config = require( '../../config/css-loader.config' );
		expect( config.importLoaders ).toEqual( 1 );
		expect( config.modules.exportLocalsConvention ).toEqual( 'camelCase' );
		expect( config.modules.localIdentName ).toEqual( 'â“‚[name]__[local]__[contenthash:base64:2]' );
		expect( config.modules.mode ).toEqual( expect.any( Function ) );
		expect( config.modules.getLocalIdent ).not.toBeDefined();
		expect( config.modules.mode )
		expect( config.sourceMap ).toEqual( true );
		expect( config.url ).toEqual( false );

		expect( config ).toMatchSnapshot( 'develop' );
	} );

	test( 'Production config', () => {
		process.env.NODE_ENV = 'production';
		const config = require( '../../config/css-loader.config' );
		expect( config.importLoaders ).toEqual( 1 );
		expect( config.modules.exportLocalsConvention ).toEqual( 'camelCase' );
		expect( config.modules.localIdentName ).toEqual( '[contenthash:base64:5]' );
		expect( config.url ).toEqual( false );
		expect( config.modules.getLocalIdent ).not.toBeDefined();

		expect( config ).toMatchSnapshot( 'production CSS modules disabled' );

		jest.resetModules();
		mockShortCssEnabled = true;
		const cssEnabled = require( '../../config/css-loader.config' );
		expect( cssEnabled.modules.getLocalIdent ).toBeDefined();
		expect( cssEnabled.modules.getLocalIdent() ).toEqual( '__TEST_CSS__' );
		expect( cssEnabled.modules.getLocalIdent() ).toEqual( getLocalIdent() );

		expect( cssEnabled ).toMatchSnapshot( 'production CSS modules enabled' );
	} );
} );
