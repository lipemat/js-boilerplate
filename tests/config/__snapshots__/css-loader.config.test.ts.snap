// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css-loader.config.test.ts Develop config: develop 1`] = `
{
  "importLoaders": 1,
  "modules": {
    "exportLocalsConvention": "camelCase",
    "localIdentName": "â“‚[name]__[local]__[contenthash:base64:2]",
    "mode": [Function],
  },
  "sourceMap": true,
  "url": false,
}
`;

exports[`css-loader.config.test.ts Production config: production CSS modules disabled 1`] = `
{
  "importLoaders": 1,
  "modules": {
    "exportLocalsConvention": "camelCase",
    "localIdentName": "[contenthash:base64:5]",
    "mode": [Function],
  },
  "url": false,
}
`;

exports[`css-loader.config.test.ts Production config: production CSS modules enabled 1`] = `
{
  "importLoaders": 1,
  "modules": {
    "exportLocalsConvention": "camelCase",
    "getLocalIdent": [MockFunction] {
      "calls": [
        [],
        [],
      ],
      "results": [
        {
          "type": "return",
          "value": "__TEST_CSS__",
        },
        {
          "type": "return",
          "value": "__TEST_CSS__",
        },
      ],
    },
    "localIdentName": "[contenthash:base64:5]",
    "mode": [Function],
  },
  "url": false,
}
`;
